library/manageiq_user.py

```diff
@@ -83,180 +101,140 @@
         """Creates the user in manageiq.
 
         Returns:
             the created user id, name, created_on timestamp,
             updated_on timestamp, userid and current_group_id
         """
+        url = '{api_url}/users'.format(api_url=manageiq.api_url)
+        resource = {'userid': userid, 'name': username, 'password': password,
+                    'group': {'id': group_id}, 'email': email}
         try:
-            url = '{api_url}/users'.format(api_url=self.api_url)
-            resource = {'userid': userid, 'name': username, 'password': password,
-                        'group': {'id': group_id}, 'email': email}
-            result = self.client.post(url, action='create', resource=resource)
-            self.changed = True
-            return dict(
-                changed=self.changed,
-                msg="Successfully created the user {userid}: {user_details}".format(userid=userid, user_details=result['results']))
+            result = manageiq.client.post(url, action='create', resource=resource)
         except Exception as e:
-            self.module.fail_json(msg="Failed to create user {userid}: {error}".format(userid=userid, error=e))
+            manageiq.module.fail_json(msg="Failed to create user {userid}: {error}".format(userid=userid, error=e))
+        self.changed = True
+        return dict(
+            changed=self.changed,
+            msg="Successfully created the user {userid}: {user_details}".format(userid=userid, user_details=result['results']))
 
-    def create_or_update_user(self, userid, username, password, group, email):
+    def create_or_update_user(self, manageiq, userid, username, password, group, email):
         """ Create or update a user in manageiq.
 
         Returns:
             Whether or not a change took place and a message describing the
             operation executed.
         """
-        group_id = self.find_group_by_name(group)
-        if not group_id:  # group doesn't exist
-            self.module.fail_json(
+        group = manageiq.find_collection_resource_by('groups', description=group)
+        if not group:  # group doesn't exist
+            manageiq.module.fail_json(
                 msg="Failed to create user {userid}: group {group_name} does not exist in manageiq".format(userid=userid, group_name=group))
 
-        user_id = self.find_user_by_userid(userid)
-        if user_id:  # user already exist
-            return self.update_user_if_required(user_id, userid, username, group_id, password, email)
+        user = manageiq.find_collection_resource_by('users', userid=userid)
+        if user:  # user already exist
+            return self.update_user_if_required(manageiq, user, userid, username, group['id'], password, email)
         else:
-            return self.create_user(userid, username, group_id, password, email)
+            return self.create_user(manageiq, userid, username, group['id'], password, email)
```

Comment:
I'd maybe give `manageiq` to the constructor, and use `self.manageiq` everywhere.
That way you don't have to pass it through from method to method.